Main Component is missing when only recurrence id exists

I get the exception "Main Component is missing" in 2.1.8. I'm sure in an earlier version maybe 2.1.3 it didn't happen.

The error is here:
https://github.com/Kozea/Radicale/blob/2.1.8/radicale/xmlutils.py#L331

I "fixed" it (not sure if fixed or hacked).
By adding this:

```diff
diff --git a/radicale/xmlutils.py b/radicale/xmlutils.py
index fd9a0d4..b8da5fe 100644
--- a/radicale/xmlutils.py
+++ b/radicale/xmlutils.py
@@ -315,6 +315,7 @@ def _visit_time_ranges(vobject_item, child_name, range_fn, infinity_fn):

     def get_children(components):
         main = None
+        rec_main = None
         recurrences = []
         for comp in components:
             if hasattr(comp, "recurrence_id") and comp.recurrence_id.value:
@@ -322,11 +323,14 @@ def _visit_time_ranges(vobject_item, child_name, range_fn, infinity_fn):
                 if comp.rruleset:
                     # Prevent possible infinite loop
                     raise ValueError("Overwritten recurrence with RRULESET")
+                rec_main = comp
                 yield comp, True, ()
             else:
                 if main is not None:
                     raise ValueError("Multiple main components")
                 main = comp
+        if main is None and len(recurrences) == 1:
+            main = rec_main
         if main is None:
             raise ValueError("Main component missing")
         yield main, False, recurrences
```

Same issue over here when trying to upload my calendar in Radicale.
```
[2021-07-28 06:31:27 +0000] [13141/Thread-9] [ERROR] '(UTC+01:00) Brussel, Kopenhagen, Madrid, Parijs'
[2021-07-28 06:31:27 +0000] [13141/Thread-9] [ERROR] '(UTC+01:00) Brussel, Kopenhagen, Madrid, Parijs'
[2021-07-28 06:31:27 +0000] [13141/Thread-9] [ERROR] '(UTC+00:00) Dublin, Edinburgh, Lisbon, London'
[2021-07-28 06:31:27 +0000] [13141/Thread-9] [ERROR] '(UTC+00:00) Dublin, Edinburgh, Lisbon, London'
[2021-07-28 06:31:27 +0000] [13141/Thread-9] [ERROR] '(UTC+01:00) Brussel, Kopenhagen, Madrid, Parijs'
[2021-07-28 06:31:27 +0000] [13141/Thread-9] [ERROR] '(UTC+01:00) Brussel, Kopenhagen, Madrid, Parijs'
[2021-07-28 06:31:27 +0000] [13141/Thread-9] [DEBUG] Rule 'fcorneli':'fcorneli' doesn't match '.+':'' from section 'root'
[2021-07-28 06:31:27 +0000] [13141/Thread-9] [DEBUG] Rule 'fcorneli':'fcorneli' matches '.+':'{user}' from section 'principal'
[2021-07-28 06:31:28 +0000] [13141/Thread-9] [ERROR] '(UTC+01:00) Brussel, Kopenhagen, Madrid, Parijs'
[2021-07-28 06:31:28 +0000] [13141/Thread-9] [ERROR] '(UTC+01:00) Brussel, Kopenhagen, Madrid, Parijs'
[2021-07-28 06:31:28 +0000] [13141/Thread-9] [ERROR] '(UTC+00:00) Dublin, Edinburgh, Lisbon, London'
[2021-07-28 06:31:28 +0000] [13141/Thread-9] [ERROR] '(UTC+00:00) Dublin, Edinburgh, Lisbon, London'
[2021-07-28 06:31:30 +0000] [13141/Thread-9] [WARNING] Bad PUT request on '/fcorneli/cf8e7878-96f6-0432-2b75-7418e047bfa8/': Main component missing
Traceback (most recent call last):
  File "/usr/local/lib64/python3.6/site-packages/radicale/app/put.py", line 73, in prepare
    item.prepare()
  File "/usr/local/lib64/python3.6/site-packages/radicale/item/__init__.py", line 394, in prepare
    self.time_range
  File "/usr/local/lib64/python3.6/site-packages/radicale/item/__init__.py", line 384, in time_range
    find_tag_and_time_range(self.vobject_item))
  File "/usr/local/lib64/python3.6/site-packages/radicale/item/__init__.py", line 262, in find_tag_and_time_range
    radicale_filter.visit_time_ranges(vobject_item, tag, range_fn, infinity_fn)
  File "/usr/local/lib64/python3.6/site-packages/radicale/item/filter.py", line 231, in visit_time_ranges
    vobject_item.vevent_list):
  File "/usr/local/lib64/python3.6/site-packages/radicale/item/filter.py", line 225, in get_children
    raise ValueError("Main component missing")
ValueError: Main component missing
[2021-07-28 06:31:30 +0000] [13141/Thread-9] [DEBUG] Response content:
Bad Request
[2021-07-28 06:31:30 +0000] [13141/Thread-9] [INFO] PUT response status for '/fcorneli/cf8e7878-96f6-0432-2b75-7418e047bfa8/' in 3.620 seconds: 400 Bad Request
```

--- a/radicale/item/filter.py
+++ b/radicale/item/filter.py
@@ -225,6 +225,7 @@ def visit_time_ranges(vobject_item: vobject.base.Component, child_name: str,
     def get_children(components: Iterable[vobject.base.Component]) -> Iterator[
                          Tuple[vobject.base.Component, bool, List[date]]]:
         main = None
+        rec_main = None
         recurrences = []
         for comp in components:
             if hasattr(comp, "recurrence_id") and comp.recurrence_id.value:
@@ -232,11 +233,14 @@ def visit_time_ranges(vobject_item: vobject.base.Component, child_name: str,
                 if comp.rruleset:
                     # Prevent possible infinite loop
                     raise ValueError("Overwritten recurrence with RRULESET")
+                rec_main = comp
                 yield comp, True, []
             else:
                 if main is not None:
                     raise ValueError("Multiple main components")
                 main = comp
+        if main is None and len(recurrences) == 1:
+            main = rec_main
         if main is None:
             raise ValueError("Main component missing")
         yield main, False, recurrences

